### DISCUSSION TITLE
StabilityDays check with no pipeline

### DISCUSSION THREAD

**Discussion by adam-moss** [renovatebot/renovate#20985] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest official docker build

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

gitlab.com EE

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Hi,

We have 

internalChecksFilter = string
stabilityDays = 3

and a couple of package rules that have automerge = true.

What we've noticed is if the repo has no pipeline (i.e. not .gitlab-ci.yml) the stability check is appears as a pipeline item, goes green, and allows the automerge to occur.

In this scenario we don't want the merge to occur, as the change hasn't been validated properly.

Is there a combination of config that will allow us to achieve that?

Thanks

### What you tried so far.

Nothing obvious is jumping out in the config.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Please clarify:
 - which version of Renovate are you running?
 - what exactly do you mean by "allows the automerge to occur" - Renovate automerge or GitLab native automerge?

Relevant: https://github.com/renovatebot/renovate/releases/tag/35.0.0

**Comment by adam-moss** [comment, marked as answer]
renovate config has automerge and platformautomerge set.

Looking at what the team has raised the reno updates are from 2 weeks ago should would have been whatever the latest official image was then.  If that's change in 35 that is what is running now so I think this is a non-issue?

**Reply by rarkins** [reply]
The change in v35 only applies to Renovate's own logic on whether to automerge. If Renovate uses GitLab native automerge, I think it's GitLab which decides "this MR looks safe to automerge", not Renovate. If I'm correct about that, then you need some different type of solution in such cases, such as "disable all Renovate status checks". But this would mean Renovate couldn't set pending stabilityDays or failed artifacts either - because if you set a pending or failed check, it would seem wrong to never set a passing check later if the situation changes.

**Reply by adam-moss** [reply]
Ok, so in this scenario I could remove the platform automerge?  That wouldn't be an issue.

**Reply by rarkins** [reply]
Yes if you remove platform automerge from such repos then Renovate should (I hope) be back in charge of whether to automerge and would not choose to do so if the only passing check was its own. Remember that in the happy case automerges will take longer though because GitLab will no longer be automerging immediately a branch goes green - instead the automerge happens on the next Renovate run after it goes green

**Reply by adam-moss** [reply]
That's ok, we run reno over night for all 10k repos and via trigger for a single repo whenever an MR is merged into it.  It's a nice solution that is working really well for us :)
